set.mm

```diff
@@ -466638,202 +466789,147 @@ real number multiplication operation (this has to be defined in the main
-        VVIVVRUYQVWAVUPWUHVVKYKZWUSUYCVUCVUAWUSVVBUYCUXSPZWJZWVCWDZVVBWUSUXSQWL
-        MZNZUYCQUXSOUOUNVFUNNZWVDAWVGVVRUYQVWAVUPWUHAUXSVOWVFVULUVNUVMZYKWUSUXR
-        VUCNZWVHWUSUXRVUAVUCWUSVVIUYQVVJWVBVVSUYQVWAVUPWUHYLVVLTAVUNVVRUYQVWAVU
-        PWUHVUOYKZWIZUXRUXSUVOZVJWVGWVHWVDQUXSUYCUVPXMZTWUSWVCUYAWUSWVCRZUXROUX
-        SWVOUXRWUSUXRVSNZWVCWUSVUCVSUXRVYNWVLXLXKYAWVOYDWUSWVCWOYPWUGVUPWUHWVCY
-        MYQWVDWVCVVBWJWVEVVBXNVVBWVCWSWVCVVBWTXAZXBWVKWBVUAUXLUYCFYEZTYJWUSUXRV
-        ONUYCWENWUSVUCVOUXRUXSUVQWVLXLUXRUVRVJUWBUVSYCYNYFYHYRWUDVWQUYOWUFWUDVW
-        QRZUYOWUFUYAVXOWUFXNWUHWUFXNUYOWUFXNWVSUYBUYDWUIVXOUYOWUFWUJXQWUKWUHUYO
-        WUFWULXQWVSUYARZVXOWUFWVTVXORZVWAWUFWWAVVTUYBUXQMZQWWAUYNUYBUXQWVTVXOWO
-        XSWWAVVIVUBWWBQPAVVIVVRUYQVWQUYAVXOVVKYKAVUBVVRUYQVWQUYAVXOAQVUCVUAAVUG
-        VUHVUIVUJVVQVWTLVUMWAVUOWBYKVUAUXLQFYETYJWUDVWQUYAVXOYMUVTYNWVSVUPRZWUH
-        WUFWWCWUHRZVWDWUOUXPWWDVWCUXRFWWDVVTUXROWWDUXRWWDVUCVSUXRVYNWWDUXRVUAVU
-        CWWDVVIUYQVVJAVVIVVRUYQVWQVUPWUHVVKYKZVVSUYQVWQVUPWUHYLZVVLTAVUNVVRUYQV
-        WQVUPWUHVUOYKZWIZXLZYAWWDYDWWDVVTWUTUYCWWDUYNUYDUXQWWCWUHWOXSWWDVVIVURW
-        VAWWEWWDUYCVUCVUAWWDWVDWVEVVBWWDWVGWVHWVDAWVGVVRUYQVWQVUPWUHWVIYKWWDWVJ
-        WVHWWHWVMVJWVNTWWDWVCUYAWWDWVCRZUXROUXSWWJUXRWWDWVPWVCWWIXKYAWWJYDWWDWV
-        CWOYPWVSVUPWUHWVCYMYQWVQXBWWGWBWVRTYJUWAXSWWDVVIUYQWUPWWEWWFWUQTYOYNYFY
-        HYRYFYNXDUYOVWKUAUXLVWEVWLUYEVWJUYNVWLUYAVWGUYDVWIUYBUXPVWEUXTUXQUWCVWL
-        UXRVWFOFUOUXPVWEUXQUWDUWEUWLUWFUWGUWHXDUAUHUXLUXLUYEUYFUYFYSUWIWFUYLAUX
-        LUXMURUYKUSUXLUXMUWJUXLBUWMUWKVDZWWKUXMUXMUXLUXLUXOUYFUWNUWOABUXNBUXNCU
-        YGAUABCEFGHIJUWPAUXNUXLUXMUFZBUXLUXMUWQAUXLBXEWWLBPAVUCBFVUEUXAUXLBUWRW
-        NUWSZWWMUWTUXBCYTNUXKUXIUXCCFBUXDMGUXEBUXJCDYTKUXJYSUXFUXGUXH $.
+         24-Sep-2023.) $)
+      cycpmcl $p |- ( ph -> ( C ` W ) e. ( Base ` S ) ) $=
+        ( cfv wf1o wcel cun c1 wceq wf syl2anc cbs crn cdif cres ccsh ccnv ccom
+        cid co cin c0 f1oi a1i cdm cc0 chash cfzo wfun cword cz 1zzd cshwf ffnd
+        wfn wf1 df-f1 sylib simprd w3a eqid cshinj syl3anc f1orn sylanbrc eqidd
+        mpi wrdf syl fdmd cshwrnid eqcomd f1oeq123d mpbird f1f1orn f1ocnv f1oco
+        wa 3syl incom disjdif eqtr3i f1oun syl22anc tocycfv uncom frnd syl5reqr
+        wss undif cvv wb fvex elsymgbas2 ax-mp sylibr ) ACCFBMZNZXFDUAMZOZAXGCF
+        UBZUCZXJPZXLUHXKUDZFQUEUIZFUFZUGZPZNZAXKXKXMNZXJXJXPNZXKXJUJZUKRZYBXRXS
+        AXKULUMAFUNZXJXNNZXJYCXONZXTAYDUOFUPMUQUIZXNUBZXNNZAXNYFVDXNUFURZYHAYFC
+        XNAFCUSOZQUTOZYFCXNSIAVAZCQFVBTVCAYJXOURZYKYIIAYCCFSZYMAYCCFVEZYNYMWGJY
+        CCFVFVGVHYLYJYMYKVIXNXNRYIXNVJCQFXNVKVPVLYFXNVMVNAYCYFXJYGXNXNAXNVOAYFC
+        FAYJYFCFSICFVQVRZVSAYGXJAYJYKYGXJRIYLQCFVTTWAWBWCAYOYCXJFNYEJYCCFWDYCXJ
+        FWEWHXJYCXJXNXOWFTYBAXJXKUJYAUKXJXKWIXJCWJWKUMZYQXKXKXJXJXMXPWLWMACXLCX
+        LXFXQABCEFGHIJWNAXLXJXKPZCXJXKWOAXJCWRYRCRAYFCFYPWPXJCWSVGWQZYSWBWCXFWT
+        OXIXGXAFBXBCXHXFDWTKXHVJXCXDXE $.
```

Comment:
Was this extreme shortening of th proof because of the changed definition/usage of cyclShift ?This would be another justification for the new definition.

Reply:
I would say it's because of the several utility theorems I proved (everything in `prop` and `snop`, `pfx1s2`, `cshwrnid` and `cshw1s2`), so it was rather broken down into smaller pieces. Which is also good, because:
- more, shorter theorems are easier to read,
- those shorter theorems are possibly reusable.

Also, the definition itself had one less bound variable, which I prefer.
So still overall it's clearly a plus.